### DISCUSSION TITLE
Does laravel automatically add the index to the primary keys?

### DISCUSSION THREAD

**Discussion by SenaJp** [laravel/framework#48491] [General] [answered]
I'm creating my tables and I would like to know if I need to add the index or not to my primary keys.

**Comment by shaedrich** [comment]
[`$table->id()`](https://laravel.com/api/10.x/Illuminate/Database/Schema/Blueprint.html#method_id) only calls [`\Illuminate\Database\Schema\Blueprint::bigIncrements()`](https://laravel.com/api/10.x/Illuminate/Database/Schema/Blueprint.html#method_bigIncrements) as `UNSIGNED BIGINT AUTO_INCREMENT`. In order to make it an index explicitly, you have to call [`\Illuminate\Database\Schema\ColumnDefinition::primary()`](https://laravel.com/api/10.x/Illuminate/Database/Schema/ColumnDefinition.html#method_primary) on the result of `$table->id()` like so:
```php
Schema::table('users', function (Blueprint $table) {
    $table->id()->primary();
});
```

However, the docs repeatedly call this
> equivalent column as a primary key:

meaning, that MySQL will implicitly add an index itself under the hood.

**Reply by henzeb** [reply]
Isn't id() not already making it a primary? According to the docs it is using bigIncrements under the hood and that does make it a primary key. https://laravel.com/docs/10.x/migrations#column-method-id

**Reply by shaedrich** [reply]
Yeah, that's actually a little confusing about the docs: It never explictly says, it sets any index with that method. However, it repeatedly states, that this is "equivalent" of it.

**Comment by iazaran** [comment, marked as answer]
Using `$table->id();` is enough to make ID, auto increment and primary key. Take a look at the DB DDL for a simple table result:
```sql
CREATE TABLE `tests` (
  `id` bigint unsigned NOT NULL AUTO_INCREMENT,
  `created_at` timestamp NULL DEFAULT NULL,
  `updated_at` timestamp NULL DEFAULT NULL,
  PRIMARY KEY (`id`),
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci
```
But for indexing, in MySQL, a primary index is automatically created, and finally, it's actually not necessary to create an index on the primary key column.

**Comment by luishenrique98** [comment]
Thank you @SenaJp for helping the community, God bless you!
